tmva/tmva/test/DNN/Utility.h

```diff
@@ -208,6 +208,28 @@ inline T relativeError(const T &x, const T &y)
   return diff / (abs(x) + abs(y));
 }
 
+//______________________________________________________________________________
+template <>
+inline Double_t relativeError(const Double_t &x, const Double_t &y)
+{
+   if ((std::abs(x) > 1e-8) && (std::abs(y) > 1e-8)) {
+      return std::fabs((x - y) / y);
+   } else {
+      return std::fabs(x - y);
+   }
+}
+
+//______________________________________________________________________________
+template <>
+inline Float_t relativeError(const Float_t &x, const Float_t &y)
+{
+   if ((std::abs(x) > 1e-8) && (std::abs(y) > 1e-8)) {
+      return std::fabs((x - y) / y);
+   } else {
+      return std::fabs(x - y);
+   }
+}
```

Comment:
Hi @simonpf, I am looking at this, and I think that the problem isn't really with the relative error calculation, but due to the fact that the relative error can indeed get quite large (~0.001). I added the test I described above, and increasing the tolerance only makes it fail less frequently, but it still fails. It may be necessary to use absolute error rather than relative error if the relative error is often large.